sw/airborne/modules/computer_vision/opencv_contour.cpp

```diff
@@ -0,0 +1,274 @@
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/computer_vision/opencv_contour.cpp"
+ * @author Peng Lu
+ * 
+ */
+
+
+#include "opencv_contour.h"
+
+
+
+using namespace std;
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+using namespace cv;
+#include "opencv_image_functions.h"
+
+//------------------------ added by Peng Lu ---------------------------//
+//#include <opencv2/highgui/highgui.hpp>
+float distance_est;
+float d_x, d_y, d_z;
+//------------------------ added by Peng Lu ---------------------------//
+
+
+
+///*
+// YUV in opencv convert to YUV on Bebop
+void yuv_opencv_to_yuv422(Mat image, char *img, int width, int height)
+{
+//Turn the opencv RGB colored image back in a YUV colored image for the drone
+  for (int row = 0; row < height; row++) {
+    for (int col = 0; col < width; col++) {
+      // Extract pixel color from image
+      cv::Vec3b &c = image.at<cv::Vec3b>(row, col);
+
+      // Set image buffer values
+      int i = row * width + col;
+      img[2 * i + 1] = c[0]; // y;
+      img[2 * i] = col % 2 ? c[1] : c[2]; // u or v
+      //img[2 * i + 1] = 0; // y;
+      //img[2 * i] = 1; // u or v
+    }
+  }
+}
+///*
+void uyvy_opencv_to_yuv_opencv(Mat image, Mat image_in, int width, int height)
+{
+//Turn the opencv RGB colored image back in a YUV colored image for the drone
+  for (int row = 0; row < height; row++) {
+    for (int col = 0; col < width; col++) {
+      // Extract pixel color from image
+      cv::Vec3b &c = image_in.at<cv::Vec3b>(row, col);
+      cv::Vec3b &c_m1 = image_in.at<cv::Vec3b>(row, col);
+      cv::Vec3b &c_p1 = image_in.at<cv::Vec3b>(row, col);
+      if (col > 0)
+      { cv::Vec3b &c_m1 = image_in.at<cv::Vec3b>(row, col-1);}
+      if (col < width)
+      { cv::Vec3b &c_p1 = image_in.at<cv::Vec3b>(row, col+1);}
+      image.at<cv::Vec3b>(row, col)[0] = c[1] ;
+      image.at<cv::Vec3b>(row, col)[1] = col % 2 ? c[0] : c_m1[0];
+      image.at<cv::Vec3b>(row, col)[2] = col % 2 ? c_p1[0] : c[0];
+
+    }
+  }
+}
+//*/
+
+
+
+int find_contour(char *img, int width, int height)
+{
+  // Create a new image, using the original bebop image.
+  Mat M(width, height, CV_8UC2, img); // original
+  //Mat M(width, height, CV_8UC3, img);
+  //Mat M(width, height, CV_8UC1, img);
+  Mat image, edge_image, thresh_image;
+
+  // 
+  //blur(image, image, Size(5, 5));
+  //GaussianBlur(image, image, Size(9,9), 5,5);
+
+  // Canny edges, only works with grayscale image
+  //int edgeThresh = 35;
+  //Canny(image, image, edgeThresh, edgeThresh * 3);
+  
```

Comment:
There is lots of commented out code, is this a work in progress or is it completed?